File: javascript/generate_random_numbers.js
const crypto = require('crypto');

// Constants
const MIN_VALUE = 1;
const MAX_VALUE = 10;
const ARRAY_LENGTH = 100;
const CHART_HEIGHT = 10;

// Generate 100 cryptographically secure random integers between 1 and 10
const numbers = Array.from({ length: ARRAY_LENGTH }, () =>
  crypto.randomInt(MIN_VALUE, MAX_VALUE + 1)
);

console.log(numbers);

// ASCII line chart using asciichart library
function plotLine(data) {
  const asciichart = require('asciichart');
  console.log(asciichart.plot(data, { 
    height: 10,
    padding: '      ',
    format: (x) => x.toFixed(1).padStart(6)
  }));
}

console.log(`\nRandom Numbers (Y: ${MIN_VALUE}-${MAX_VALUE}, X: 1-${ARRAY_LENGTH}):`);
plotLine(numbers);

File: python/generate_random_numbers.py
import secrets
import asciichartpy

# Constants
ARRAY_LENGTH = 100
MIN_VALUE = 1
MAX_VALUE = 10
CHART_HEIGHT = 10

# Generate 100 cryptographically secure random integers between 1 and 10
numbers = [secrets.randbelow(MAX_VALUE) + MIN_VALUE for _ in range(ARRAY_LENGTH)]

# Print the numbers
print(numbers)

# Plot each number along the x-axis, y-axis is the value (1 to 10)
print(f"\nRandom Numbers (Y: {MIN_VALUE}-{MAX_VALUE}, X: 1-{ARRAY_LENGTH}):")
print(asciichartpy.plot(numbers, {'height': CHART_HEIGHT}))
print(f"X-axis: 1 to {ARRAY_LENGTH} (index of each random number)")
print(f"Y-axis: Value ({MIN_VALUE} to {MAX_VALUE})")
